src/book.py

```diff
@@ -180,39 +217,79 @@ def tokenize(self, on, rem_stopwords=True, stopword_lang='english',
+        rem_stopwords: Same as in Tokenize
+        ret_as_array: (Default = False): Returns one BookText object by appending all groups of snippets, else returns
+                                         an array of BookText objects
+        with_replace: (Default = True): For non-contiguous case, makes sure there is no repetition. If turned off, different
+                                        snippets in a group might have repeated char/word/sent.
+        random_seed: When positive, generates the same snippets everytime
+        
         """
-        # TODO add ability for random seed
+        if (random_seed != 0):
+            random.seed(random_seed)
+            
         if 'char' in on.lower():
-            assert length < len(self._text)
+            tokens = self._text
         elif 'word' in on.lower():
-            tokens = word_tokenize(self._text)
-            assert length < len(tokens)
+            tokens = self.tokenize('word', rem_stopwords)
         elif 'sent' in on.lower():
-            tokens = sent_tokenize(self._text)
+            tokens = self.tokenize('sent', rem_stopwords)
         else:
             raise KeyError(
-                "Arugument 'on' must refer to character, word, or sentence")
-
-        if non_cont:
-            # TODO add functionality for non-continuous random sampling
-            pass
+                "Argument 'on' must refer to character, word, or sentence")
+            
+        assert groups*length <= len(tokens)
+        return_array = []
+
+        if randomized:
+            start = random.randint(0, len(tokens) - groups*length)
+            tokens = tokens[start:]
         else:
-            if 'char' in on.lower():
-                start = random.randint(0, len(self._text) - length)
-                snippet = self._text[start:start + length]
-            elif 'word' in on.lower():
-                start = random.randint(0, len(tokens) - length)
-                snippet = (" ").join(tokens[start:start + length])
-            elif 'sent' in on.lower():
-                start = random.randint(0, len(tokens) - length)
-                snippet = (" ").join(tokens[start:start + length])
+            start = 0
+        
+        if non_cont == False:
+            for gr in range(groups):
+                snippet = BookText(rawtext=''.join((" ").join(tokens[length*gr:length*(gr+1)])), 
+                               author = self.author, title = self.title, meta = self.meta)
+                return_array.append(snippet)
+        else:
+            for gr in range(groups):
+                random_sample_of_indices = sorted(random.sample(range(0, len(tokens)), length), reverse=False)
+                dummy_string = str()
+                for num in random_sample_of_indices:
+                    dummy_string += tokens[num] + str(' ')
+                dummy_string.rstrip()    
+                snippet = BookText(rawtext= dummy_string, author = self.author, 
```

Comment:
I believe for the sake of pep8 compliance the spaces between all `=` should be removed.

Reply:
Ok. Thanks. 'Twas an oversight.